app/models/supplier.rb

```diff
@@ -6,26 +6,35 @@ class Supplier < ActiveRecord::Base
   # save lists in an array in database
   serialize :lists
 
+  # @rails4: enum ftp_type: {bnn: 0, foodsoft: 1}
+  FTP_TYPES = {bnn: 0, foodsoft: 1}
+
   EMAIL_RE = /^([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})$/i.freeze
 
   validates :name, :address, :phone, presence: true
-  validates :bnn_host, :bnn_user, :bnn_password, :bnn_sync, presence: true, if: :bnn_sync?
+  validates :ftp_host, :ftp_user, :ftp_password, :ftp_sync, presence: true, if: :ftp_sync?
+  validates :ftp_type, presence: true, inclusion: {in: FTP_TYPES.values}, if: :ftp_sync?
   validates :mail_from, presence: true, format: { with: EMAIL_RE }, if: :mail_sync?
   validates :mail_type, inclusion: { in: ArticleImport.file_formats.keys }, presence: true, if: :mail_sync?
 
-  scope :bnn_sync, ->{ where(bnn_sync: true) }
+  scope :ftp_sync, ->{ where(ftp_sync: true) }
   scope :mail_sync, ->{ where(mail_sync: true) }
 
   before_create :create_salt
 
-  def bnn_path
+  def ftp_path
+    # path should read "ftp_files", but has "bnn_files" for historical reasons
     Rails.root.join("supplier_assets", "bnn_files", id.to_s)
   end
 
   def mail_path
     Rails.root.join("supplier_assets", "mail_files", id.to_s)
   end
 
+  def ftp_type_string
+    FTP_TYPES.key(ftp_type).to_s
+  end
```

Comment:
Perhaps I'd rather make this something like:
```ruby
def ftp_type
  FTP_TYPES.key(super)
end
```

You can still access the `attributes` if you really need the integer.

Or put the string in the database (that's what I usually do, it makes it more obvious what it is, and there is not a need for high performance here).

Reply:
I changed it to the string approach with d664a26e99ac139785366d982b3d9b9de56faf5b. Let me rebase the whole branch before we merge.